core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/plangen/LogicalPlan.java

```diff
@@ -103,24 +111,90 @@ public LogicalPlan(
      * @param operatorID, the ID of an operator
      * @return Schema, which includes the attributes setting of the operator
      */
-    public Schema getOperatorOutputSchema(String operatorID) throws PlanGenException {
+    public Schema getOperatorOutputSchema(String operatorID) throws PlanGenException, DataflowException {
 
         if (UPDATED) {
```

Comment:
This isn't related to Ming's PR, but I don't remember this 'UPDATED' variable here before. When did it get introduced?

Reply:
Also, I see that this function is called inside tests, where the buildQueryPlan() will make UPDATED false. So, the code within UPDATED block won't be called ever.

Reply:
The UPDATED tag is mainly used as a reference for the incremental changes feature that we planning to introduce later.
It states that when any of the graph changes or when the graph is newly created, only those part that is under if (UPDATED) needed to be re-executed. 

Sorry for the confusion, this part should be in another branch, I will delete it